modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -47,48 +58,89 @@ const onSetupToggle = (editor: Editor, selections: Selections, formatName: strin
   };
 };
 
-const applyTableCellStyle = <T extends Item>(editor: Editor, style: string) => (item: T): void => {
-  editor.execCommand('mceTableApplyCellStyle', false, { [style]: item.value });
+export const isListGroup = (item: UserListItem): item is UserListGroup =>
+  Obj.hasNonNullableKey(item as Record<string, any>, 'menu');
+
+const buildListItems = (items: UserListItem[]): Dialog.ListBoxItemSpec[] =>
+  Arr.map(items, (item) => {
+    // item.text is not documented - maybe deprecated option we can delete??
+    const text = item.text || item.title;
+    if (isListGroup(item)) {
+      return {
+        text,
+        items: buildListItems(item.menu)
+      };
+    } else {
+      return {
+        text,
+        value: item.value
+      };
+    }
+  });
+
+const buildMenuItems = (
+  editor: Editor,
+  selections: Selections,
+  items: UserListItem[],
+  format: string,
+  onAction: (value: string) => void
+): Menu.NestedMenuItemContents[] =>
+  Arr.map(items, (item): Menu.NestedMenuItemContents => {
+    // item.text is not documented - maybe deprecated option we can delete??
```

Comment:
Can we Jira this up, or delete it before we merge?

Reply:
We can't delete until 6.0, so logging a Jira wouldn't quite make sense yet. I'll add it to the 6.0 initiatives list in Notion instead.